Containers/Camera/Camera.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { Camera, Permissions, FileSystem } from 'expo';
+
+export default class CameraExample extends Component {
+  constructor() {
+    super();
+    this.state = {
+      hasCameraPermission: null,
+      type: Camera.Constants.Type.back,
+    }
+  };
+
+  componentWillMount = async () => {
+    const { status } = await Permissions.askAsync(Permissions.CAMERA);
+    this.setState({ hasCameraPermission: status === 'granted' });
+  };
+
+  takePicture = () => {
+    if (this.camera) {
+      this.camera.takePictureAsync({ onPictureSaved: this.onPictureSaved });
+    }
+  };
+
+  onPictureSaved = async (photo) => {
+    Expo.FileSystem.copyAsync
+    const newLocation = `${FileSystem.documentDirectory}photos/${Date.now()}.jpg`
+    await FileSystem.moveAsync({
+      from: photo.uri,
+      to: newLocation
+    });
+    this.props.savePicture(newLocation);
```

Comment:
Taking the temporary location (uri) and the moving it to a location that the app will store it. The location is then sent to App state via savePicture

Reply:
Impressive!